assets/controllers/markdown_toolbar_controller.js

```diff
@@ -0,0 +1,39 @@
+// SPDX-FileCopyrightText: 2023-2024 /kbin & Mbin contributors
+//
+// SPDX-License-Identifier: AGPL-3.0-only
+
+import { Controller } from '@hotwired/stimulus';
+
+/* stimulusFetch: 'lazy' */
+export default class extends Controller {
+    addSpoiler(event) {
+        event.preventDefault();
+
+        const input = document.getElementById(this.element.getAttribute('for'));
+        let spoilerBody = '_____';
+        let contentAfterCursor;
+
+        const start = input.selectionStart;
+        const end = input.selectionEnd;
+
+        const contentBeforeCursor = input.value.substring(0, start);
+        if (start === end) {
+            contentAfterCursor = input.value.substring(start);
+        } else {
+            contentAfterCursor = input.value.substring(end);
+            spoilerBody = input.value.substring(start, end);
+        }
+
+        const spoiler = `
+::: spoiler spoiler
```

Comment:
```suggestion
::: spoiler spoiler title
```
I think this would encourage the user to replace this title. I was confused at first as to where the title comes from

Reply:
this might still be confusing maybe, since now people might remove the whole line.. Including the first "spoiler" text,  I dunno. The current `spoiler spoiler`  seem to be a safe default.

Reply:
Or we combine spoiler title together or something like `spoiler-title`

Reply:
`spoiler-title` would good :+1: